src/dead_code_elimination.cpp

```diff
@@ -44,9 +44,10 @@ void dead_code_elimination::apply(module& m) const
         // Skip the last instruction
         if(i == last)
             break;
-        // Skip instruction with empty shape as output unless its a builtin or undefined or identity
+        // Skip instruction with empty shape as output unless its a builtin, undefined, identity, or
+        // allocate
         if(i->get_shape().elements() == 0 and i->name().front() != '@' and
-           i->name() != "undefined" and i->name() != "identity")
+           i->name() != "undefined" and i->name() != "identity" and i->name() != "allocate")
```

Comment:
Why do we skip `allocate`?

Reply:
we aren't skipping, the first condition was passing, so it was then skipping. I had to add an extra condition because of that.

Reply:
Essentially there are allocates where the number of elements is 0 (from what I saw this was coming from MIOpen convolution), so we need to still remove them as dead code.

Reply:
I see, we should probably use `contains({"undefined", "identity", "allocate"}, i->name())`.

Reply:
changed